frappe/public/js/frappe/form/grid.js

```diff
@@ -838,7 +838,9 @@ export default class Grid {
 					acc[d.fieldname] = d.default;
 					return acc;
 				}, {});
-				this.df.data.push({ idx: this.df.data.length + 1, __islocal: true, ...defaults });
+				let idx = this.df.data.length + 1;
```

Comment:
`idx` is already used as a parameter, so redefining it here might cause issues. Also, this can be a `const`.